packages/api/src/controllers/animalGroupController.js

```diff
@@ -0,0 +1,141 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Model, transaction } from 'objection';
+import baseController from './baseController.js';
+import AnimalGroupModel from '../models/animalGroupModel.js';
+import AnimalGroupRelationshipModel from '../models/animalGroupRelationshipModel.js';
+import AnimalBatchGroupRelationshipModel from '../models/animalBatchGroupRelationshipModel.js';
+import AnimalModel from '../models/animalModel.js';
+
+const animalGroupController = {
+  getFarmAnimalGroups() {
+    return async (req, res) => {
+      try {
+        const { farm_id } = req.headers;
+        const rows = await AnimalGroupModel.query()
+          .where({ farm_id })
+          .whereNotDeleted()
+          .withGraphFetched({ animal_relationships: true, batch_relationships: true });
+        return res.status(200).json(
+          rows.map(({ animal_relationships, batch_relationships, ...rest }) => ({
```

Comment:
I would just plug the idea that when you keep the structure of the graph and related table it can make reading the data as well as the code manipulation much easier.

Reply:
Yeah here I still had to do this mapping because the related graph was getting me something like `related_batches: [{ id: 5 }, { id: 7}]` which is fine if I wanted more properties than just an ID but seemed overkill for that

Reply:
Yeah as long as its a "join like" table thats probably fine. If we put base properties on it in the future though then it makes more sense. 

Eg with deleted and updated and created property we can track when an animal entered the group and left the group.